### ISSUE TITLE
A11y keyboard focus control in the Progress Bar Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3710]
#### Bug, feature request, or proposal:
Bug: Missing element text label.

#### What is the expected behavior?
All page elements are expected to have an associated text label.

#### What is the current behavior?
Currently the Slider element text label is not associated with the Slider control.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the "Configurable progress-bar" demo component in the Tab Panel.
JAWS says "Primary radio button checked 1 of 3", and the braille display shows 
"<x> Primary rbtn 1 or 3".
Note, this demo uses two Radio Button groups "Color:" and "Mode:", in which the Arrow keys rotate between the grouped Buttons.
- Selecting a color does not invoke an automatic announcement of the "Result" change in the Progress Bar.
- Selecting a Mode does not invoke an automatic announcement of the "Result" change in the Progress Bar (should this be expected?).
- Selecting Indeterminate or Query will cause the Tab key to move the focus to the page footer without any "Result" message announcement (Is this to be expected?).
- Selecting Determinate will cause the Tab key to move the focus to the progress md-slider control, but the word "Progress:" is not announced as it is not associated with the md-slider element.
- Selecting Buffer will cause the Tab key to move the focus to the progress md-slider control and then to the Buffer md-slider, but in either case the word "Progress:" and "Buffer:" is not announced as they are not associated with the md-slider element.

2. Press the Tab key to move focus to the md-slider control.
When "Determinate" or "Buffer" is selected, focus moves to the md-slider control.
JAWS says "Lefgt right slider 50 to increase or decrease use the arrow keys", and the braille display shows "lrtrk 50".
Note, this control behaves as expected when using Arrows, Home, End, PgUP, and PgDn keys.
However, the user does not know which Slider control has focus, as the Label element is not associated with the md-slider control.

**Comment by jelbourn** [comment]
These all appear to be bugs with the demo rather than the underlying component. 

We may want to log a warning when progress-bar is missing an appropriate label attribute.
